pkgs/development/python-modules/versioningit/default.nix

```
@@ -0,0 +1,33 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, importlib-metadata
+, packaging
+, setuptools
+, tomli
+}:
+
+buildPythonPackage rec {
+  pname = "versioningit";
+  version = "1.1.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-O3VlotuS8SBrW4M6csxpNsinTWXpQiTwqetUeZ0VYdk=";
+  };
+
+  format = "pyproject";
+
+  propagatedBuildInputs = [
+    importlib-metadata
+    packaging
+    setuptools
+    tomli
+  ];
```

```suggestion
  propagatedBuildInputs = [
    packaging
    setuptools
    tomli
  ] ++ lib.optionals (pythonOlder "3.10") [
    importlib-metadata
  ];
```

`dataclasses` is mentioned for Python < 3.7. I suggest to use `disabled = pythonOlder "3.7";` as everything else is EOL.

Could you please check if we can use `pytestCheckHook` to run the tests?

---

Okay, I was able to get the tests working properly (except some that want to write to Nix store)